addons/udes_stock/models/abstract_stock_model.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+_CREATE_OBJECTS = ['group', 'package']
+
+class AbstractStockModel(models.AbstractModel):
+    _name = 'abstract.stock.model'
+
+    def _validate_get_args(self, obj, identifier, create):
+        """ Validates the identifier and create in terms of type
+        :args:
+            - obj: str
+                If 'group' only query name else query both name and barcode
+            - identifier: str/int
+                Thing to look for
+        :kwargs:
+            - create: Boolean
+                Only allowed to be true if obj is in _CREATE_OBJECTS and identifier is a string
+        :returns:
+
+        NOTE: This restricts only new groups being created, this means
+            that a str identifier, which could be a barcode, can not be used to
+            create a new product
+        """
+        if not isinstance(identifier, (str, int)):
+            raise ValidationError(_(f'Identifier must be either int or str, not {identifier}'))
+        elif not isinstance(create, bool):
+            raise ValidationError(_(f'Create option must be a boolean, not {type(identifier)}'))
+        elif isinstance(identifier, int) and create:
+            raise ValidationError(_(f'Cannot create a new {obj} with integer identifier'))
+        elif isinstance(identifier, str) and create and obj not in _CREATE_OBJECTS:
+            raise ValidationError(_(f'A new {obj} cannot be created!'))
+
+    def _get_domain(self, obj, identifier):
+        """ Get domain to search
+        The identifier should first be verified
+        :args:
+            - obj: str
+                If obj in _CREATE_OBJECTS only query name else query both name and barcode
+            - identifier: str/int
+                Thing to look for
+        :returns:
+            domain
+
+        """
+        if isinstance(identifier, int):
+            return [('id', '=', identifier)]
+        else:
+            if obj in _CREATE_OBJECTS:
```

Comment:
Can we have a list of "str" fields that by default only contains name, and then in other models we change it to also contain 'barcode' ?